wandb/wandb_run.py

```diff
@@ -43,19 +45,48 @@ class Run(object):
     def __init__(self, run_id=None, mode=None, dir=None, group=None, job_type=None,
                  config=None, sweep_id=None, storage_id=None, description=None, resume=None,
                  program=None, args=None, wandb_dir=None, tags=None, name=None, notes=None):
-        # self.id is actually stored in the "name" attribute in GQL
+        # self.storage_id is "id" in GQL.
+        self.storage_id = storage_id
+        # self.id is "name" in GQL.
         self.id = run_id if run_id else util.generate_id()
-        self.display_name = self.id
+        # self._name is  "display_name" in GQL.
+        self._name = None
```

Comment:
Having `self.name` and `self._name` is a little confusing.  I don't have a great suggestion to remedy.  I assume it exists because we need to know the difference between a wandb.ai generated display_name and one that we specify ourselves?

Reply:
They're the same. I think this sort of thing is pretty common, having a hidden state that's exposed by a property with the same name. I think the main reason to have it here is so we can update `_name_and_description` whenever it changes. It'd be good to get rid of the whole description thing but I can't think of an easy/clean way to do it yet.

Reply:
I'm used to seeing `_variable_name` exposed in an actual `@property` method.  The fact we also have a `name` attribute feels confusing.